url6dc6a3bff8_JasonxJiang_PokemonEvolutionaryAlgorithm_tgz-pJ8-ee_ea_pkg1_EEEA1J8/src/ee/ea/pkg1/pokemon/KyuremW.java

```diff
@@ -0,0 +1,191 @@
+            IVSPD = 31;
+            IVSPEED = 31;
+            EVHP = 6;
+            EVATK = 0;
+            EVDEF = 0;
+            EVSPA = 252;
+            EVSDF = 0;
+            EVSPEED = 252;
+            statBoostATK = 1;
+            statBoostDEF = 1;
+            statBoostSPA = 1;
+            statBoostSDF = 1;
+            statBoostSPEED = 1;
+            level = 100;
+            finalHp = (int) ((((IVHP + (2 * baseHealth) + (EVHP / 4)) * level) / 100) + 10);
+            finalAtk = (int) (((((IVATK + (2 * baseAttack) + (EVATK / 4)) * level) / 100) + 5) * natureAtk);
+            finalDef = (int) (((((IVDEF + (2 * baseDefense) + (EVDEF / 4)) * level) / 100) + 5) * natureDef);
+            finalSpa = (int) (((((IVSPA + (2 * baseSpecialAttack) + (EVSPA / 4)) * level) / 100) + 5) * natureSpa);
+            finalSdf = (int) (((((IVSPD + (2 * baseSpecialDefense) + (EVSDF / 4)) * level) / 100) + 5) * natureSdf);
+            finalSpeed = (int) (((((IVSPEED + (2 * baseSpeed) + (EVSPEED / 4)) * level) / 100) + 5) * natureSpeed);
+            name = "Kyurem-W";
+            primary = new Dragon();
+            secondary = new Ice();
+            moveLibrary = new HashMap<String, Move>();
+            Move m1 = new DragonPulse();
+            Move m2 = new FocusBlast();
+            Move m3 = new EarthPower();
+            Move m4 = new FusionFlare();
+            Move m5 = new IceBeam();
+            Move m6 = new Blizzard();
+            Move m7 = new ShadowBall();
+            Move m8 = new DracoMeteor();
+            moveLibrary.put("dragon pulse", m1);
+            moveLibrary.put("focus blast", m2);
+            moveLibrary.put("earth power", m3);
+            moveLibrary.put("fusion flare", m4);
+            moveLibrary.put("ice beam", m5);
+            moveLibrary.put("blizzard", m6);
+            moveLibrary.put("shadow ball", m7);
+            moveLibrary.put("draco meteor", m8);
+            potentials = new ArrayList<String>();
+            potentials.add("dragon pulse");
+            potentials.add("focus blast");
+            potentials.add("earth power");
+            potentials.add("fusion flare");
+            potentials.add("ice beam");
+            potentials.add("blizzard");
+            potentials.add("shadow ball");
+            potentials.add("draco meteor");
+            circumvent = new ArrayList<String>();
+            battleReady = new HashMap<String, Move>();
+            for (int i = 0; i < 4; i++) {
+                int randomIndex = (int) (Math.random() * potentials.size());
+                String key = potentials.get(randomIndex);
+                while (circumvent.contains(key)) {
+                    int nextRandom = (int) (Math.random() * potentials.size());
+                    key = potentials.get(nextRandom);
+                }
+                battleReady.put(key, moveLibrary.get(key));
+                circumvent.add(key);
+                //potentials.remove(randomIndex);
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 96:106, 177:187.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.